lib/code_corps/github/event/installation/repos.ex

```diff
@@ -141,18 +134,28 @@ defmodule CodeCorps.GitHub.Event.Installation.Repos do
   @spec create(GithubAppInstallation.t, map) :: {:ok, GithubRepo.t}
   defp create(%GithubAppInstallation{} = installation, %{} = repo_attributes) do
     %GithubRepo{}
-    |> Changeset.change(repo_attributes)
+    |> changeset(repo_attributes)
     |> Changeset.put_assoc(:github_app_installation, installation)
     |> Repo.insert()
   end
 
   @spec update(GithubRepo.t, map) :: {:ok, GithubRepo.t}
   defp update(%GithubRepo{} = github_repo, %{} = repo_attributes) do
     github_repo
-    |> Changeset.change(repo_attributes)
+    |> changeset(repo_attributes)
     |> Repo.update()
   end
 
+  @spec changeset(GithubRepo.t, map) :: Changeset.t
+  defp changeset(%GithubRepo{} = github_repo, %{} = repo_attributes) do
+    github_repo
+    |> Changeset.change(repo_attributes)
+    |> Changeset.validate_required([
+      :github_id, :name, :github_account_id,
+      :github_account_avatar_url, :github_account_login, :github_account_type
+    ])
```

Comment:
Instead of having the adapter implicitly doing validation of the repo payload, I switched to simply having the process (and the transaction) erroring out if there are any validation errors on any of the repos.